src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -292,6 +292,31 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     return result
   }
 
+  /** Returns KDoc comments including those of primary constructor parameters. */
+  private fun kdocWithConstructorParameters(): CodeBlock {
+    if (primaryConstructor == null || primaryConstructor.parameters.isEmpty()) {
+      return kdoc
+    }
+
+    val constructorProperties = constructorProperties()
+
+    if (primaryConstructor.parameters.all { it.kdoc.isEmpty() } &&
```

Comment:
I think this check is redundant, given that we check for empty KDoc in the snippet below. Is this to avoid doing the `toBuilder()`/`build()` on the KDoc in case there's nothing to add?

Reply:
Correct, but it's probably premature optimization. I'll pull it out.